tests/plugins/chart/basic_chart.test.ts

```diff
@@ -1446,6 +1448,216 @@ describe("Chart design configuration", () => {
+            B4: { content: "12" },
+            B5: { content: "13" },
+            B6: { content: "14" },
+            B7: { content: "15" },
+            B8: { content: "16" },
+            B9: { content: "17" },
+            C1: { content: "second column dataset" },
+            C2: { content: "31" },
+            C3: { content: "32" },
+            C4: { content: "33" },
+            C5: { content: "34" },
+            C6: { content: "21" },
+            C7: { content: "22" },
+            C8: { content: "23" },
+            C9: { content: "24" },
+          },
+        },
+      ],
+    });
+  });
+
+  test("One dataset: all data complete", () => {
+    createChart(aggregatedModel, aggregatedChart, "42");
+    let chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: true });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { type: "line" });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: false });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { type: "pie" });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
+
+    updateChart(aggregatedModel, "42", { aggregated: true });
+    chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([24, 26, 28, 30]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4"]);
+  });
+
+  test("One dataset: some (not all) values are lost for a label", () => {
+    createChart(aggregatedModel, aggregatedChart, "42");
+    let chart = (aggregatedModel.getters.getChartRuntime("42") as BarChartRuntime).chartJsConfig;
+    expect(chart.data!.datasets![0].data).toEqual([10, 11, 12, 13, 14, 15, 16, 17]);
+    expect(chart.data!.labels).toEqual(["P1", "P2", "P3", "P4", "P1", "P2", "P3", "P4"]);
```

Comment:
don't repeat this everywhere.
Keep tests small and focused on the behavior under test.
The normal aggregate is already tested above